providers-sdk/v1/inventory/inventory.go

```diff
@@ -417,7 +417,7 @@ func (cfg *Config) Clone(opts ...CloneOption) *Config {
 	}
 
 	clonedObject := proto.Clone(cfg).(*Config)
-
+	clonedObject.Id = 0
```

Comment:
we aren't relying on the `Id` to be copied anywhere, where `Clone` is called. That's why I opt to make it default that the ID isn't copied. If in the future we have cases that we need to copy the ID too, we can add an option to clone the ID too and fix it with that:
```go
clone := inv.Clone(WithId())
```